go/src/lib/README.md

```diff
@@ -0,0 +1,100 @@
+Summary:
+    Provides shared common APIs, like log related
+    Provides APIs for clients to communicate with Engine
+    Unit tests
+
+
+lomcommon:
+    Provides APIs for
+        GetLogLevel:
+            Returns current log level
+
+        SetLogLevel
+            Accepts the log level to set
+            Only messages with log level <= set log level will be logged.
+            Default log level is syslog.LOG_ERR
+            With log level set to syslog.LOG_DEBUG, all messages are printed
+            to STDOUT too, to faciliate debugging.
+           
+        LogPanic
+            Log messages with syslog.LOG_CRIT and call for process exit.
+
+        LogError
+            Log messages with syslog.LOG_ERR
+
+        LogWarning
+            Log messages with syslog.LOG_WARNING
+
+        LogInfo
+            Log messages with syslog.LOG_INFO
+
+        LogDebug
+            Log messages with syslog.LOG_DEBUG
+            
+    
+    All LoM code (plugins, pluginMgr, Engine, Config-Mgr, ...) uses this single set of APIs for
+    logging. This helps us provide a unified presentation of LoM logs and ability to tweak any
+    in one place for entire LoM
+
+TODO:
+    Add facility for callers to override syslog provided Writer with any. 
+    This is an *advanced* use case and hence we do, when we see a need.
+
+
+lomipc:
+    Provides APIs for clients to use to contact Engine
```

Comment:
Are we persisting the outstanding requests anywhere ? If not, when engine crashes for some reason, how do we ensure plugin manager is also restarted, so that both engine and plugin manager are in-sync with each other.

Reply:
engine is a critical process. It's crash will restart entire service.